src/main/java/se/liu/ida/hefquin/engine/queryproc/impl/poptimizer/cardinality/VarSpecificCardinalityEstimationImpl.java

```diff
@@ -165,6 +165,35 @@ protected CompletableFuture<Integer> _initiateUnionCardinalityEstimation(
 
 	}
 
+	protected CompletableFuture<Integer> _initiateMultiwayUnionCardinalityEstimation(
+			final List<PhysicalPlan> subPlans,
+			final Var v )
+	{
+		final Set<Var> allJoinVars = ExpectedVariablesUtils.intersectionOfAllVariables( subPlans.toArray(new PhysicalPlan[0]) );
+
+		if ( allJoinVars.contains(v) ) {
```

Comment:
I am not sure but I believe that this is wrong. Variable `v` is a join variable not only if it appears in _all_ of the subplans under the union but also if it appears in only one of these subplans.